src/applications/disability-benefits/all-claims/utils/submit.js

```diff
@@ -498,6 +580,15 @@ export function transformTreatmentFacilities(
 }
 
 export const addForm0781V2 = formData => {
+  // If a user selected any workflow option outside of submitting the online form,
+  // we want to remove 0781-related data
+  if (
+    formData.mentalHealthWorkflowChoice !==
+    form0781WorkflowChoices.COMPLETE_ONLINE_FORM
+  ) {
+    delete0781FormData(formData);
+  }
```

Comment:
I think this might be better positioned after the conditional that follows it because, for the new 0781, nothing should happen to the `formData` unless `syncModern0781Flow` is true.